databroker/experimental/server_ext.py

```diff
@@ -125,6 +127,54 @@ def put_data(self, body):
         if result.matched_count != result.modified_count:
             raise ValueError("Error while writing to database")
 
+    def put_metadata(self, metadata, specs):
+
+        revisions_doc = (
+            self.revisions.find({"key": self.doc.key}).sort({"revision": -1}).limit(1)
+        )
+        if revisions_doc is not None:
+            revision = int(revisions_doc["revision"]) + 1
+        else:
+            revision = 1
+
+        validated_revision = DocumentRevision(
+            key=self.doc.key,
+            structure_family=self.doc.structure_family,
+            structure=self.doc.structure,
+            metadata=self.doc.metadata,
+            specs=self.doc.specs,
+            mimetype=self.doc.mimetype,
+            created_at=self.doc.created_at,
+            data_url=self.doc.data_url,
+            updated_at=self.doc.updated_at,
+            revision=revision,
+        )
+
+        self.revisions.insert_one(validated_revision.dict())
+
+        updated_at = datetime.now(tz=timezone.utc)
+        self.doc.updated_at = updated_at
+
+        if len(metadata) > 0:
```

Comment:
If I want to update `metadata` to be empty `{}` or `specs` to be empty `[]`, shouldn't that update be processed?

Reply:
Fixed in the latest commit